### PR TITLE
BAEL-1080 Introduction to Future and Pattern Matching in Vavr

### PR DESCRIPTION
**Pull request by hunghoang**
I've closed the last pull request and open another one by removing all Pattern Matching Unit test and add  some unit test of Future with Pattern Matching

### CODE REVIEW

File: spring-all/src/main/java/org/baeldung/bean/injection/Author.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.baeldung.bean.injection;
```
Comment (pivovarit) [inline spring-all/src/main/java/org/baeldung/bean/injection/Author.java]: Please remove files that are not related to the article

File: vavr/src/test/java/com/baeldung/vavr/future/FutureUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+            .intValue());
+    }
+
+    @Test
+    public void givenAFuture_WhenCallCancel_ShouldStopTheActionOfFuture() {
+        long waitTime = 1000;
+        int[] store = new int[1];
+        Future<Integer> future = Future.of(() -> {
+            Thread.sleep(waitTime);
+            store[0] = 1;
```
Comment (pivovarit) [inline vavr/src/test/java/com/baeldung/vavr/future/FutureUnitTest.java]: It should be possible to obtain the result through Future's API and not by doing this hack

**Comment by hunghoang** [comment]
create another one
